src/app/nav-bar/nav-bar.component.ts

```diff
@@ -0,0 +1,143 @@
+import { ChangeContext } from '@angular-slider/ngx-slider';
+import { AfterViewChecked, ChangeDetectorRef, Component, ElementRef, ViewChild } from '@angular/core';
+import { BehaviorSubject, combineLatest } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+import { AppConfig } from '../app.config';
+import { EVTModelService } from '../services/evt-model.service';
+import { EVTStatusService } from '../services/evt-status.service';
+
+@Component({
+  selector: 'evt-nav-bar',
+  templateUrl: './nav-bar.component.html',
+  styleUrls: ['./nav-bar.component.scss'],
+})
+export class NavBarComponent implements AfterViewChecked {
+  @ViewChild('thumbnailsContainer') thumbnailsContainer: ElementRef;
+
+  private currentPageInfo$ = combineLatest([
+    this.evtModelService.pages$,
+    this.evtStatusService.currentPage$,
+  ]);
+
+  prevNavigationDisabled$ = this.currentPageInfo$.pipe(
+    map(([pages, page]) => pages.findIndex((p) => p.id === page.id) === 0),
+  );
+  nextNavigationDisabled$ = this.currentPageInfo$.pipe(
+    map(([pages, page]) => pages.findIndex((p) => p.id === page.id) === pages.length - 1),
+  );
+
+  currentPage$ = this.currentPageInfo$.pipe(
+    map(([pages, page]) => pages.findIndex((p) => p.id === page.id)),
+  );
+
+  thViewerSettings$ = new BehaviorSubject({ col: 1, row: 1 });
+
+  thumbnailsButton = AppConfig.evtSettings.ui.thumbnailsButton;
+  thumbnailsPanelOpened$ = new BehaviorSubject(false);
+
+  viscollButton = AppConfig.evtSettings.ui.viscollButton;
+  viscollPanelOpened$ = new BehaviorSubject(false);
+
+  navigationDisabled$ = combineLatest([this.thumbnailsPanelOpened$, this.viscollPanelOpened$]).pipe(
+    map(([thumbnailsPanelOpened, viscollPanelOpened]) => thumbnailsPanelOpened || viscollPanelOpened),
+  );
+
+  pageSliderOptions$ = combineLatest([this.navigationDisabled$, this.evtModelService.pages$])
+    .pipe(
+      map(([navigationDisabled, pages]) => ({
+        floor: 0,
+        ceil: pages.length - 1,
+        showSelectionBar: true,
+        translate: (value: number): string => pages[value]?.label ?? '',
+        disabled: navigationDisabled,
+      })),
+    );
+
+  constructor(
+    private evtStatusService: EVTStatusService,
+    private cdref: ChangeDetectorRef,
+    public evtModelService: EVTModelService,
+  ) {
+  }
+
+  ngAfterViewChecked() {
+    this.calculateThumbsPerPage();
```

Comment:
What is it for and why?

Reply:
The component used to show thumbnails accepts in input col and rows in order to handle thumbs list pagination (and lighten the loading of files). Since I've decided to make the thumb dimensions customizable with scss variables, and I cannot access scss from ts, the only way to know the height and width of thumb and calculate the maximum elements per row and the maximum rows is to wait the rendering of element. If you have a better idea, tell me! :)